### DISCUSSION TITLE
Alternative to current price per unit presentation

### DISCUSSION THREAD

**Discussion by ipaleka** [asastats/channel#140] [General]
A question has been asked by @kerrilija in Discord about the mechanism for a unit price on the left:

> Yldy price per unit shows 476, while on DEXes it is around 500? Actually, the difference is significant for most coins. Where is that information pulled from?

We use the available DExes to get that unit price on the left, we send 1 unit to the engine and it calculates and returns the value in ALGO that we then present to the user. We use a thousand times smaller unit If such a value crosses 100 ALGO, like for wrapped coins (goBTC, goETH, ...). Bear in mind that everything on the left side has only an informational purpose and those data are never used by the engine for any calculation.

In the quoted example, YLDY has got the best price on AlgoDEX so we have used its buy order price as a price per unit.

If someone finds some other approach as a better fit, please present it here in this thread.

**Comment by ipaleka** [comment]
by kerrilija in Discord

- - - - -
For me using the price from a pool that has got the biggest ASA-Algo liquidity sounds like the most logical approach.
- - - - -

**Reply by ipaleka** [reply]
> For me using the price from a pool that has got the biggest ASA-Algo liquidity sounds like the most logical approach.

Using some kind of mean or average from all of them can probably prevent some of the issues a single source can have (like if it has got the biggest liquidity, but that's just 25% of all liquidity).

P.S. dunno, really, a price per unit will always be controversial as it's a construct, as opposed to the right side that gives the exact values.

**Reply by kerrilija** [reply]
The average could also give a wrong representation of the value. For example, HumbleSwap has a liquidity of 1.18K Algos in YLDY pool while Tinyman has around 1.5M. In this case, I would pop Humble out of the equation.

My suggestion is that, if liquidity difference between DEXes is big(80, 90% or more like in this case) the smaller liquidity pool should be ignored.

**Reply by ipaleka** [reply]
by kerrilija

- - - - -

Let it be a user setting then.

- - - - -


[Added!](https://github.com/asastats/channel/wiki/FeaturesDependentOnUserSettings#preferred-price-per-unit-algorithm)

**Reply by ipaleka** [reply]
> The average could also give a wrong representation of the value. For example, HumbleSwap has a liquidity of 1.18K Algos in YLDY pool while Tinyman has around 1.5M. In this case, I would pop Humble out of the equation.

Of course, in this case we'd be calculating `0.00118*x + 1.5*y` as a starting point.
